packages/curve-ui-kit/src/themes/components/slider/utils.ts

```diff
@@ -129,13 +140,35 @@ export const getOrientationConfig = (orientation?: SliderProps['orientation']):
   return orientationConfigMap[key] ?? orientationConfigMap[DEFAULT_ORIENTATION]
 }
 
-export const borderedRailBackground = (design: DesignSystem, direction: 'to right' | 'to top') => {
+export const borderedRailBackground = (
+  design: DesignSystem,
+  direction: 'to right' | 'to top',
+  sectionCount: number = BORDERED_SECTION_COUNT,
+) => {
   const segment = design.Color.Primary[200]
   const line = design.Color.Neutral[500]
-  const segments = `linear-gradient(${direction}, ${segment} 0%, ${segment} 25%, ${segment} 25%, ${segment} 50%, ${segment} 50%, ${segment} 75%, ${segment} 75%, ${segment} 100%)`
-  const borders = `repeating-linear-gradient(${direction}, transparent 0, transparent calc(25% - 1px), ${line} calc(25% - 1px), ${line} 25%)`
+
+  // Calculate the percentage for each section
+  const sectionPercentage = 100 / sectionCount
+
+  // Build the segments gradient stops dynamically
+  const segmentStops = Array.from({ length: sectionCount }, (_, i) => {
```

Comment:
I just [moved](https://github.com/curvefi/curve-frontend/blob/e938242ded0a0d39b6ff023b77321f62e79706e5/packages/prices-api/src/objects.util.ts#L37-L40) the cypress helper to reuse in #1559 so we can get rid of this ugly syntax :grin: